src/lib.rs

```diff
@@ -121,11 +123,14 @@ pub fn edit_distance(search_term: &str, known_term: &str) -> usize {
 
     // Run the algorithm.
     for i in 1..n {
+        let mut known_chars = known_term.chars();
+        let search_char = search_chars.next().unwrap();
         for j in 1..m {
-            let mut sub_cost = 1;
-            if search_chars[i - 1] == known_chars[j - 1] {
-                sub_cost = 0;
-            }
+            let sub_cost = if search_char == known_chars.next().unwrap() {
+                0
+            } else {
+                1
+            };
```

Comment:
Also, I don't believe that the array indexing version of this section and the code that you've replaced it with actually compare the same elements. Notice that we are indexing the array at `i - 1` and not `i` which I believe is what your code is doing. I'm not 100% sure about this since I don't know too much about how `.next()` works on strings.

Reply:
know_chars is a chars iterator, it actually points to index 0 of your string, when j = 1, know_chars.next() yield the first char, when j = m - 1, know_chars.next() yield the last chars, aka str[m - 2], noticed `m = search_term.len() + 1`, so the code logic should be the same.

Reply:
Ah okay, so if that's the same then this change is fine by me. How would you plan to access `i - 2` in the transposition? Maybe save it in a temporary variable before you call `.next()`?

Reply:
This impletation is wrong,we don't need to talk about the detail for now, let me think other way to optimize this.